crates/cranelift/src/obj.rs

```diff
@@ -101,6 +101,12 @@ pub struct ObjectBuilder<'a> {
     pending_relocations: Vec<(u64, &'a [Relocation])>,
     windows_unwind_info: Vec<RUNTIME_FUNCTION>,
     systemv_unwind_info: Vec<(u64, &'a systemv::UnwindInfo)>,
+
+    // The unwind info _must_ come directly after the text section. Our FDE's
+    // instructions are encoded to rely on this placement. We use this `bool`
+    // for debug assertions to ensure that we get the ordering correct.
+    #[cfg(debug_assertions)]
```

Comment:
FWIW I don't think it's worth doing the `debug_assertions` dance here, I'd just make everything `assert!` to unconditionally check.